### PR TITLE
Agile 6.6.0.12 CSVM Support

### PR DESCRIPTION
**Pull request by ghost**
Added Support for latest Agile CSVM 

This supports Agile 6.6.0.12 CSVM 

Except Opcode LdvirtFtn as i cannot seem to get it to save correct handler

### CODE REVIEW

**Comment by 0xd4d** [comment]
> Except Opcode LdvirtFtn as i cannot seem to get it to save correct handler

Do you mean you can't get the compiler to create a ldvirtftn for you so you can get the handler or that Agile.NET doesn't use that handler? The compiler will use a ldvirtftn in the first `M` method.

```C#
using System;

namespace ConsoleApp16
{
    class Program
    {
        static void Main(string[] args)
        {
        }

        static void M(Program p)
        {
            EventHandler h = p.Handler;
            M(h);
        }

        static void M(EventHandler h) { }

        public virtual void Handler(object sender, EventArgs e) { }
    }
}
```

File: de4dot.code/deobfuscators/Agile_NET/vm/v2/CsvmResources.resx [hunk 1]
```diff
@@ -112,12 +112,12 @@
     <value>2.0</value>
   </resheader>
   <resheader name="reader">
-    <value>System.Resources.ResXResourceReader, System.Windows.Forms, Version=2.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089</value>
+    <value>System.Resources.ResXResourceReader, System.Windows.Forms, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089</value>
```
Comment (0xd4d) [inline de4dot.code/deobfuscators/Agile_NET/vm/v2/CsvmResources.resx]: Undo this

File: de4dot.code/deobfuscators/Agile_NET/vm/v2/CsvmResources.resx [hunk 2]
```diff
@@ -112,12 +112,12 @@
...
   </resheader>
   <resheader name="writer">
-    <value>System.Resources.ResXResourceWriter, System.Windows.Forms, Version=2.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089</value>
+    <value>System.Resources.ResXResourceWriter, System.Windows.Forms, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089</value>
```
Comment (0xd4d) [inline de4dot.code/deobfuscators/Agile_NET/vm/v2/CsvmResources.resx]: Undo this

File: de4dot.code/deobfuscators/Agile_NET/vm/v2/CsvmResources.resx [hunk 3]
```diff
@@ -112,12 +112,12 @@
...
   </resheader>
-  <assembly alias="System.Windows.Forms" name="System.Windows.Forms, Version=2.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089" />
+  <assembly alias="System.Windows.Forms" name="System.Windows.Forms, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089" />
```
Comment (0xd4d) [inline de4dot.code/deobfuscators/Agile_NET/vm/v2/CsvmResources.resx]: Undo this

**Review by ghost — CHANGES_REQUESTED** [review]

**Comment by 0xd4d** [comment]
Abandoned PR

**Comment by CreateAndInject** [comment]
How to generate CSVMx.bin？ I want to do it for 6.0.4.31.

**Comment by ElektroKill** [comment]
I would also like to know how to generate a csvm.bin file.

**Pull request closed without merging**